llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -943,9 +943,12 @@ bool LoopVectorizationLegality::canVectorizeInstrs() {
         VecCallVariantsFound = true;
 
       // Check that the instruction return type is vectorizable.
+      // We can't vectorize casts from vector type to scalar type.
       // Also, we can't vectorize extractelement instructions.
       if ((!VectorType::isValidElementType(I.getType()) &&
            !I.getType()->isVoidTy()) ||
+          (isa<CastInst>(&I) &&
```

Comment:
Nit: I don't think the ampersand is required; s/&I/I/.

Reply:
Thanks, will fix.